braindecode/models/reve.py

```diff
@@ -0,0 +1,618 @@
+        Number of frequencies for the Fourier positional embedding.
+    patch_size : int
+        Size of each patch for patch embedding.
+    patch_overlap : int
+        Overlap size between patches.
+
+    References
+    ----------
+    .. [reve] El Ouahidi, Y., Lys, J., Thölke, P., Farrugia, N., Pasdeloup, B., Gripon, V., Jerbi, K. & Lioi, G. (2025).
+        REVE: A Foundation Model for EEG - Adapting to Any Setup with Large-Scale Pretraining on 25,000 Subjects. NeurIPS.
+        Online: `https://arxiv.org/abs/2510.21585`
+    """
+
+    def __init__(
+        self,
+        n_outputs=None,
+        n_chans=None,
+        chs_info=None,
+        n_times=None,
+        input_window_seconds=None,
+        sfreq=None,
+        # REVE specific parameters
+        embed_dim=512,
+        depth=22,
+        heads=8,
+        head_dim=64,
+        mlp_dim_ratio=2.66,
+        use_geglu=True,
+        freqs=4,
+        patch_size=200,
+        patch_overlap=20,
+        attention_pooling: bool = False,
+    ):
+        super().__init__(
+            n_outputs=n_outputs,
+            n_chans=n_chans,
+            chs_info=chs_info,
+            n_times=n_times,
+            input_window_seconds=input_window_seconds,
+            sfreq=sfreq,
+        )
+
+        self.embed_dim = embed_dim
+        self.freqs = freqs
+        self.patch_size = patch_size
+        self.overlap_size = patch_overlap
+        self.depth = depth
+        self.heads = heads
+        self.head_dim = head_dim
+        self.mlp_dim_ratio = mlp_dim_ratio
+        self.use_geglu = use_geglu
+
+        self.use_attention_pooling = attention_pooling
+
+        self.to_patch_embedding = patch_embedding(self.embed_dim, self.patch_size)
+
+        self.fourier4d = FourierEmb4D(self.embed_dim, freqs=self.freqs)
+        self.mlp4d = mlp_pos_embedding(self.embed_dim)
+        self.ln = nn.LayerNorm(self.embed_dim)  # 4DPE module layernorm
+
+        self.transformer = TransformerBackbone(
+            dim=self.embed_dim,
+            depth=self.depth,
+            heads=self.heads,
+            head_dim=self.head_dim,
+            mlp_dim=int(self.embed_dim * self.mlp_dim_ratio),
+            geglu=self.use_geglu,
+        )
+
+        final_dim = self._get_flattened_output_dim()
+
+        if self.use_attention_pooling:
+            self.final_layer = nn.Sequential(
+                nn.LayerNorm(self.embed_dim),
+                nn.Linear(self.embed_dim, self.n_outputs),
+            )
+        else:
+            self.final_layer = nn.Sequential(
+                nn.Flatten(),
+                nn.LayerNorm(final_dim),
+                nn.Linear(final_dim, self.n_outputs),
+            )
+
+        self.cls_query_token = nn.Parameter(torch.randn(1, 1, self.embed_dim))
+
+        self._position_bank = RevePositionBank()
+
+        self.default_pos = None
+        if chs_info is not None:
+            self.default_pos = self.get_positions([ch["ch_name"] for ch in chs_info])
```

Comment:
Why not directly using the positions provided in chs_info? Where did the positions come from in your channel bank?
Something like this:
```suggestion
            self.default_pos = self.get_positions([ch["loc"][:3] for ch in chs_info])
```

Or maybe a hybrid solution such that we also cover the case where the channel names are non-informative but positions are provided

Reply:
The positions have been manually realigned to be consistent across montages. It's still possible to provide raw positions through the `pos` argument of the forward, but the recommended way to get the 3D coordinates is to use the `get_positions`, that takes a list of channel names as an input.